src/NetTopologySuite/Index/Strtree/AbstractSTRtree.cs

```diff
@@ -368,8 +371,16 @@ private static bool RemoveItem(AbstractNode<T, TItem> node, TItem item)
             for (var i = node.ChildBoundables.GetEnumerator(); i.MoveNext(); )
             {
                 var childBoundable = i.Current as ItemBoundable<T, TItem>;
-                if (childBoundable != null && ReferenceEquals(childBoundable.Item, item))
+                if (childBoundable != null)
+                {
+                    // EqualityComparerForRemoveItem might be null, to signal that we should use the
+                    // default equality comparer for the type.
+                    if (EqualityComparerForRemoveItem?.Equals(childBoundable.Item, item) ??
+                        EqualityComparer<TItem>.Default.Equals(childBoundable.Item, item))
```

Comment:
Are you sure that ??-ing to the default comparer for value types for **each** `childBoundable` is actually faster than setting default comparer within `InitializeEqualityComparerForRemoveItem`?

Reply:
Yes, but it probably doesn't matter here.  I'll replace this with the more obvious version.